### PR TITLE
fix: update dep tree and remove version from dep name

### PR DESCRIPTION
**Pull request by aryehbeitz**
**Done**
* fixed dep tree
* remove version from package name in tree
* fixed `from` to include the current dep

### CODE REVIEW

File: lib/formats/dotnet-cli-parser.js [hunk 1]
```diff
@@ -19,9 +20,8 @@ function collectFlatList(targetObj) {
 function registerDependency(packageTree, resolvedName) {
   var version = resolvedName.split('@')[1];
```
Comment (michael-go) [inline lib/formats/dotnet-cli-parser.js]: better switch between these 2 lines

File: lib/formats/dotnet-cli-parser.js [hunk 2]
```diff
@@ -73,6 +72,8 @@ function parse(fileContent, packageTree) {
   // restructure tree with "from" chains.
   function buildPath(node, from) {
     node.from = node.from.concat(from);
+    node.from = _.uniq(node.from);
```
Comment (adrukh) [inline lib/formats/dotnet-cli-parser.js]: Why is this needed? Why would the from path collected so far contain duplicates?

File: package.json [hunk 3]
```diff
@@ -31,6 +32,7 @@
     "zip": "^1.2.0"
   },
   "devDependencies": {
+    "lodash-cli": "^4.17.4",
```
Comment (adrukh) [inline package.json]: Is this how we did this in the php plugin? I think the built lib should not be checked in to git, but should be published to npmjs. This requires another key in package.json?

File: lib/formats/dotnet-cli-parser.js [hunk 4]
```diff
@@ -45,19 +45,18 @@ function parse(fileContent, packageTree) {
 
     Object.keys(targetObj).forEach(function (key) {
       var resolvedName = convertFromPathSyntax(key);
-      var dependency = packageTree.dependencies[resolvedName];
+      var dependency = packageTree.dependencies[
+        resolvedName.split('@').shift()];
```
Comment (michael-go) [inline lib/formats/dotnet-cli-parser.js]: too complex for a one liner, do a simple split()[0] on a separate line

File: package.json [hunk 5]
```diff
@@ -3,6 +3,7 @@
   "description": "![logo](https://res.cloudinary.com/snyk/image/upload/v1468845259/logo/snyk-dog.svg) ## Snyk: NuGet Plugin ***",
   "main": "lib/index.js",
   "scripts": {
+    "build": "./node_modules/.bin/lodash -p -o ./dist/lodash-min.js include=uniq",
```
Comment (michael-go) [inline package.json]: not sure about that ...

File: lib/formats/dotnet-cli-parser.js (hunk 2 above)
Comment (aryehbeitz) [reply lib/formats/dotnet-cli-parser.js]: strange to me too. when I add line 76 to include the current dep, it causes the previous dep to duplicate.
apparently the `forEach` is the culprit and I thought it cleaner to add `uniq` than `if` statements.
your thoughts?

File: package.json (hunk 3 above)
Comment (aryehbeitz) [reply package.json]: in the php plugin, the dist folder is included
not sure what to do here

File: package.json (hunk 5 above)
Comment (aryehbeitz) [reply package.json]: we did it in the php plugin, but no excuse ;)
what do you suggest?

File: lib/formats/dotnet-cli-parser.js (hunk 2 above)
Comment (adrukh) [reply lib/formats/dotnet-cli-parser.js]: You are using concat, but Why are you not using `node.from.push(node.name + '@' + node.version)`?

Not that is explains the duplication, but let's not solve this with `uniq`. Let's get to the bottom of this. Add a test that reproduces this duplication over a fixture manifest file, use `devtool` to debug?

File: package.json (hunk 5 above)
Comment (adrukh) [reply package.json]: You did something different on the php plugin - the `dist` folder is built on travis and pushed to npmjs, not checked in to git.
Not an issue really because we won't need this once the duplication issue is resolved?

File: lib/formats/dotnet-cli-parser.js (hunk 2 above)
Comment (aryehbeitz) [reply lib/formats/dotnet-cli-parser.js]: I had to add the final `from` to the end of the recursive function

**Comment by aryehbeitz** [comment]
@adrukh this is ready; can we merge?
